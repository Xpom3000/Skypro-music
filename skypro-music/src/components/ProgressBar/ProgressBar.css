/* ProgressBar.module.css */
.styledProgressInput {
    --progress-height: 5px;
    --progress-color: #b672ff;
    --progress-bg-color: #2e2e2e;
  
    margin: 0;
    width: 100%;
    height: var(--progress-height);
    -webkit-appearance: none;
    appearance: none;
    cursor: pointer;
    background: transparent;
    position: relative;
    overflow: hidden;
    transition: height 0.2s;
  
    background: linear-gradient(to right, var(--progress-color) var(--value), var(--progress-bg-color) var(--value));
  }
  
  .styledProgressInput:hover {
    --progress-height: 8px;
  }
  
  /* Стили для WebKit браузеров */
  .styledProgressInput::-webkit-slider-runnable-track {
    height: var(--progress-height);
    background: var(--progress-bg-color);
  }
  
  .styledProgressInput::-webkit-slider-thumb {
    -webkit-appearance: none;
    appearance: none;
    width: 25px;
    height: 25px;
    border-radius: 50%;
    background: var(--progress-color);
  }
  
  /* Стили для Firefox */
  .styledProgressInput::-moz-range-track {
    background: var(--progress-bg-color);
    height: var(--progress-height);
  }
  
  .styledProgressInput::-moz-range-thumb {
    border: none;
    height: 25px;
    width: 25px;
    border-radius: 50%;
    background: var(--progress-color);
  }