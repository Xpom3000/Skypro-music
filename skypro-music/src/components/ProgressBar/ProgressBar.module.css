.styledProgressInput {
  --progress-height: 5px;
  --progress-color: #b672ff;
  --progress-bg-color: #2e2e2e;

  margin: 0;
  width: 100%;
  height: var(--progress-height);
  -webkit-appearance: none;
  appearance: none;
  cursor: pointer;
  background: transparent;
  position: relative;
  overflow: hidden;
  transition: height 0.2s;

  background: linear-gradient(to right, var(--progress-color) var(--value), var(--progress-bg-color) var(--value));
}

.styledProgressInput:hover {
  --progress-height: 8px;
}

/* Стили для WebKit браузеров */
.styledProgressInput::-webkit-slider-runnable-track {
  position: relative;
  height: var(--progress-height);
  background: var(--progress-bg-color);
}

.styledProgressInput::-webkit-slider-thumb {
  --thumb-height: 1px;
  --thumb-width: 1px;
  position: relative;
  -webkit-appearance: none;
  width: var(--thumb-width, var(--thumb-height));
  box-shadow: calc(-100vmax - var(--thumb-width, var(--thumb-height))) 0 0
    100vmax var(--progress-color);
}

/* Стили для Firefox */
.styledProgressInput::-moz-range-track {
  width: 100%;
  height: var(--progress-height);
  background: var(--progress-bg-color);
  border: none;
  border-radius: 0px;
}


.styledProgressInput::-moz-range-thumb {
  border: none;
  height: 25px;
  width: 25px;
  border-radius: 50%;
  background: var(--progress-color);
}

.styledProgressInput::-moz-range-progress {
  background-color: var(--progress-color);
  height: var(--progress-height);
}