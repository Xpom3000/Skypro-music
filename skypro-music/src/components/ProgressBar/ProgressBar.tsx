import { ChangeEvent } from "react";
import styles from "./ProgressBar.module.css";

type ProgressBarType = {
    max: number | undefined; 
    value: number;
    step: number;
    onChange: (e:ChangeEvent<HTMLInputElement>) => void;
}

export default function ProgressBar({ max=0, value, step, onChange }:ProgressBarType) {
  return (
    <input
	    className={styles.styledProgressInput} // Применение стилей к ползунку
	    type="range" // Тип элемента - ползунок
	    min="0" // Минимальное значение ползунка
	    max={max} // Максимальное значение, зависит от длительности аудио
	    value={value} // Текущее значение ползунка
	    step={step} // Шаг изменения значения
	    onChange={onChange} // Обработчик события изменения
	  />
  );
}