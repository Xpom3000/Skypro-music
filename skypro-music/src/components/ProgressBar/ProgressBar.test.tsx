import renderer  from"react-test-renderer";
import ProgressBar from "./ProgressBar";

describe("Компонент ProgressBar", () => {
  it("Успешный рендер ProgressBar", () => {
    const handleSeek = jest.fn();
    const test = renderer
      .create(
        <ProgressBar
          max={135.444898}
          value={2.377663}
          step={0.01}
          onChange={handleSeek}
        />
      )
      .toJSON();
    expect(test).toMatchSnapshot();
  });
});