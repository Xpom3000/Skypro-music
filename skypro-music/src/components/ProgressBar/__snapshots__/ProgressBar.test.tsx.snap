// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент ProgressBar Успешный рендер ProgressBar 1`] = `
<input
  className="styledProgressInput"
  max={135.444898}
  min="0"
  onChange={[MockFunction]}
  step={0.01}
  type="range"
  value={2.377663}
/>
`;
